tests/test_compare_objects.py

```diff
@@ -0,0 +1,12 @@
+import tests
+from pyVmomi import vim
+from operator import eq as equals
+
+
+class CompareTests(tests.VCRTestBase):
+
+    def test_compare_objects(self):
+        compare_str = "Test"
+        compare_obj = vim.Datastore
+        self.assertFalse(equals(compare_str, compare_obj))
+        self.assertTrue(equals(compare_obj, compare_obj))
```

Comment:
Looks like vim.Datastore is not ManagedObject and the __eq__ is not used here.

>>> dataobj = vim.Datastore
>>> from pyVmomi.VmomiSupport import ManagedObject
>>> isinstance(dataobj, ManagedObject)
False

Since the result only can be True when we comparing two managed object, maybe we can just check if the other object is ManagedObject? If not, we can return False. Otherwise, we compare other fields.